crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -147,6 +161,310 @@ async fn generate_cac(
+    key_val: Value,
+    override_id: &str,
+) -> Vec<(Context, Map<String, Value>, Value, String)> {
+    let mut res: Vec<(Context, Map<String, Value>, Value, String)> = Vec::new();
+    for ct in contexts {
+        let ct_dimensions = extract_dimensions(&ct.condition).unwrap();
+        if ct_dimensions.contains_key("variantIds") {
+            continue;
+        }
+        let override_keys = &ct.override_with_keys;
+        if override_keys.contains(&override_id.to_owned()) {
+            res.push((
+                ct,
+                overrides.clone(),
+                key_val.clone(),
+                override_id.to_string(),
+            ));
+        }
+    }
+    res
+}
+
+fn construct_new_payload(req_payload: &Map<String, Value>) -> web::Json<PutReq> {
+    let mut res = req_payload.clone();
+    res.remove("to_be_deleted");
+    res.remove("override_id");
+    res.remove("id");
+    if let Some(Value::Object(res_context)) = res.get("context") {
+        if let Some(Value::Object(res_override)) = res.get("override") {
+            return web::Json(PutReq {
+                context: res_context.to_owned(),
+                r#override: res_override.to_owned(),
+            });
+        }
+    }
+    web::Json(PutReq {
+        context: Map::new(),
+        r#override: Map::new(),
+    })
+}
+
+async fn reduce_context_key(
+    user: User,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    mut og_contexts: Vec<Context>,
+    mut og_overrides: Map<String, Value>,
+    check_key: &str,
+    dimension_schema_map: &HashMap<String, (JSONSchema, i32)>,
+    default_config: Map<String, Value>,
+    is_approve: bool,
+) -> Config {
+    let default_config_val = default_config.get(check_key);
+    let mut contexts_overrides_values: Vec<(Context, Map<String, Value>, Value, String)> =
+        Vec::new();
+
+    for (key, val) in og_overrides.clone() {
+        match val {
+            Value::Object(obj_val) => {
+                if let Some(ans_val) = obj_val.get(check_key) {
+                    let mut temp_obj_val = obj_val.clone();
+                    temp_obj_val.remove(check_key);
+                    let context_arr = get_contextids_from_overrideid(
+                        og_contexts.clone(),
+                        temp_obj_val,
+                        ans_val.clone(),
+                        &key,
+                    );
+                    contexts_overrides_values.extend(context_arr);
+                }
+            }
+            _ => (),
+        }
+    }
+
+    contexts_overrides_values.sort_by(|a, b| {
+        (validate_dimensions_and_calculate_priority(
+            "context",
+            &(b.0).condition,
+            dimension_schema_map,
+        )
+        .unwrap())
+        .cmp(
+            &(validate_dimensions_and_calculate_priority(
+                "context",
+                &(a.0).condition,
+                dimension_schema_map,
+            )
+            .unwrap()),
```

Comment:
Can we avoid the raw unwrap()

Reply:
Hey, in both unwrap above
I'm not able to propagate the errors or return it because it's inside the closure 
What i can do is create a new array first, which has all the priorities and then sort it 
Should i go ahead with that ?

Reply:
yeah that would be better.

Reply:
Can you make the closure return a Result type? You can then propagate errors through that?